data_steward/cdr_cleaner/cleaning_rules/deid/repopulate_person_controlled_tier.py

```diff
@@ -180,6 +190,60 @@ class JoinExpression(NamedTuple):
 WHERE o.rank_order = 1
 """)
 
+ETHNICITY_QUERY_TEMPLATE = JINJA_ENV.from_string("""
+SELECT DISTINCT
+    o.* EXCEPT (rank_order)
+    {% if translate_source_concepts is not none and translate_source_concepts|length > 0 -%}{{'\t'}}   
+    REPLACE(
+        CASE {{prefix}}_source_concept_id
+        {% for translate_source_concept in translate_source_concepts %}
+            WHEN {{translate_source_concept.concept_id}} THEN {{translate_source_concept.translated_concept_id}}
+            {%- if translate_source_concept.comment is not none %} /*{{translate_source_concept.comment}}*/{%- endif %}{{'\n'}}
+        {%- endfor %}
+            ELSE {{prefix}}_concept_id
+        END AS {{prefix}}_concept_id
+    )
+    {% endif %}
+FROM 
+(
+    SELECT
+        p.person_id,
+        COALESCE(ethnicity_ob.value_as_concept_id,
+            CASE race_ob.value_source_concept_id
+                WHEN  {{no_matching_concept_id}} THEN {{no_matching_concept_id}} /*missing answer*/
```

Comment:
Post discussion notes:
1. Update to IF statement to have the same behavior as RT.

Reply:
Just added the change.